src/applications/login/containers/SignInApp.jsx

```diff
@@ -0,0 +1,233 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import AlertBox from '@department-of-veterans-affairs/formation-react/AlertBox';
+
+import ExternalServicesError from 'platform/monitoring/external-services/ExternalServicesError';
+// import recordEvent from 'platform/monitoring/record-event';
+import SubmitSignInForm from 'platform/static-data/SubmitSignInForm';
+import { login, signup } from 'platform/user/authentication/utilities';
+import environment from 'platform/utilities/environment';
+
+import downtimeBanners from '../utilities/downtimeBanners';
+
+const loginHandler = loginType => () => {
+  // TODO add separate login tracking for this page
+  //   recordEvent({ event: `login-attempted-${loginType}` });
+  login(loginType, 'v1');
+};
+
+const handleDsLogon = loginHandler('dslogon');
+const handleMhv = loginHandler('mhv');
+const handleIdMe = loginHandler('idme');
```

Comment:
These floating strings always bothered me. There isn't a list of constants anywhere we can use to add some meaning and help show the relationship across files?

Reply:
it looks like `src/platform/monitoring/DowntimeNotification/config/externalServices.js` has all these defined, more from the perpective of "configs" for the Downtime Notifications. Would it make sense to move or redefine these elsewhere for more shared use?

Reply:
Good question...it seems like they could be used here and https://github.com/department-of-veterans-affairs/vets-website/pull/11599/files#diff-ac40956ca0948c728e184a47b617d8dcR28, but it's hard to say if the `externalServices` values really have a relationship with the IDP values or if they should be kept separate. It might be worth asking how these values are represented in the API, but I don't know unfortunately.